### ISSUE TITLE
Error in diagnosing list of designs

### ISSUE THREAD

**Issue by macartan** [DeclareDesign/DeclareDesign#308]
When below  is attempted design_2 is not appearing in output. Reason is possibly related to the fact that "Term" is used in design_1 but not in design_2.

```
design_1 <- simple_factorial_designer(N=500, outcome_means = c(0,0,1,2), w_A = 0, w_B = 0)
design_2 <- multi_arm_designer(N=500, m = 3, means = c(0,0,1))

diagnose_design(design_1, 
                design_2,
                sims = 10)
```

**Comment by acoppock** [comment]
this works:

```
diagnose_design(list(design_1, design_2), sims = 10)
```

**Comment by macartan** [comment]
@acoppock  that does not work for me! only design_1 is in the diagnosis

The issue I think is in places like this:

```
diagnosands_df <- mapply(calculate_diagnosands_single_design, simulations_list,
        MoreArgs = list(
          diagnosands = diagnosands,
          group_by_set = group_by_set
        ),
        SIMPLIFY = FALSE
      )

```
Here `group_by_set` includes `term` but because some of the designs  do not have `term` the term column in simulations_list for them is all NA and they got dropped entirely in the grouping 

I think this could be fixed by placing something like this right at the top of diagnose_design after group_by_set is defined:

```
simulations_df[,group_by_set][is.na(simulations_df[, group_by_set])]  <- "NA"
```

Though perhaps there are neater ways. A disadvantage of this approach is if there really is a set of groups in which some units have term = NA and some have term = "NA" then these will get conflated

**Comment by acoppock** [comment]
should be fixed in #318

**Comment by graemeblair** [comment]
Fixed!

```
Research design diagnosis based on 10 simulations. Diagnosand estimates with bootstrapped standard errors in parentheses (100 replicates).

 Design Label  Estimand Label Estimator Label Term N Sims   Bias   RMSE  Power Coverage
     design_1           ate_A  No_Interaction    A     10   0.50   0.51   1.00     0.00
                                                          (0.04) (0.03) (0.00)   (0.00)
     design_1           ate_B  No_Interaction    B     10   0.52   0.52   1.00     0.00
                                                          (0.02) (0.02) (0.00)   (0.00)
     design_1     interaction     Interaction  A:B     10  -0.02   0.26   1.00     0.80
                                                          (0.08) (0.05) (0.00)   (0.13)
     design_2 ate_Y_Z_2_Y_Z_1             DIM <NA>     10   0.00   0.12   0.00     1.00
                                                          (0.03) (0.01) (0.00)   (0.00)
     design_2 ate_Y_Z_3_Y_Z_1             DIM <NA>     10  -0.00   0.11   1.00     1.00
                                                          (0.03) (0.02) (0.00)   (0.00)
     design_2 ate_Y_Z_3_Y_Z_2             DIM <NA>     10  -0.01   0.09   1.00     1.00
                                                          (0.03) (0.02) (0.00)   (0.00)
 Mean Estimate SD Estimate Mean Se Type S Rate Mean Estimand
          1.50        0.12    0.09        0.00          1.00
        (0.04)      (0.02)  (0.00)      (0.00)        (0.00)
          0.52        0.07    0.09        1.00          0.00
        (0.02)      (0.01)  (0.00)      (0.00)        (0.00)
          0.98        0.27    0.18        0.00          1.00
        (0.08)      (0.06)  (0.00)      (0.00)        (0.00)
          0.00        0.13    0.11         NaN          0.00
        (0.03)      (0.02)  (0.00)          NA        (0.00)
          1.00        0.12    0.11        0.00          1.00
        (0.03)      (0.02)  (0.00)      (0.00)        (0.00)
          0.99        0.10    0.11        0.00          1.00
        (0.03)      (0.02)  (0.00)      (0.00)        (0.00)
```
